File: Character Search/Pre Solutions/CharacterSearch.py
def find_unique_character(string: str) -> str:
    chars = string.lower()
    copies = []
    size = len(chars)

    for target_index in range(size):
        frequency = 0
        current = chars[target_index]
        index = -1

        if current in copies:
            index = 1

        if index >= 0:
            continue

        for next_index in range(target_index + 1, size):
            if current == chars[next_index]:
                copies.append(current)
                frequency = frequency + 1
                break

        if frequency < 1:
            return chars[target_index]

    return '{} has no unique character'.format(chars)


result = find_unique_character('PurpleStar')
print(result)

File: Character Search/Pre Solutions/CharacterSearch.cpp
#include <iostream>
#include <algorithm> 
#include<cmath>
#include<fstream>
#include <string> 
#include <cctype>
#include <cctype>
#include <cctype>
#include <cctype>
#include <cctype>
#include <cctype>

using namespace std;

string lower(string str);
string find_unique_character(string str);

int main() {
	string result = find_unique_character("PopularStar");
	cout << result << endl;
}

static string find_unique_character(string str) 
{	
	string chars = lower(str);
	string copies;

	int size = chars.size();
	int frequency;
	char current;

	for (int target_index = 0; target_index < size; ++target_index) {
		frequency = 0;
		current = chars.at(target_index);

		int index = copies.find(current);

		if (index >= 0) {
		    continue;
		}

		for (int next = target_index + 1; next < size; ++next) {
		     if (current == chars.at(next)) {
			 copies += current;
			 ++frequency;
		         break;
		     }
		}
		
		if (frequency < 1) {
		    return chars.substr(target_index, 1);
		}
	}

	return chars.append(" has no unique character");
}

static string lower(string str) {
	transform(str.begin(), str.end(), str.begin(), [](unsigned char ch) {
		return tolower(ch);
	});

	return str;
}
